api/v1beta1/linodemachine_types.go

```diff
@@ -0,0 +1,425 @@
+
+	// Image is the Linode image to use for the instance.
+	// +optional
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
+	Image string `json:"image,omitempty"`
+
+	// BackupsEnabled is a boolean indicating whether backups should be enabled for the instance.
+	// +optional
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
+	BackupsEnabled bool `json:"backupsEnabled,omitempty"`
+
+	// PrivateIP is a boolean indicating whether the instance should have a private IP address.
+	// +optional
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
+	PrivateIP *bool `json:"privateIP,omitempty"`
+
+	// NetworkHelper is an option usually enabled on account level. It helps configure networking automatically for instances.
+	// You can use this to enable/disable the network helper for a specific instance.
+	// For more information, see https://techdocs.akamai.com/cloud-computing/docs/automatically-configure-networking
+	// Defaults to true.
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
+	// +optional
+	NetworkHelper *bool `json:"networkHelper,omitempty"`
+
+	// Tags is a list of tags to apply to the Linode instance.
+	// +optional
+	// +listType=set
+	Tags []string `json:"tags,omitempty"`
+
+	// TOOD: should we add AdditionalUserData here like rke2 does? See https://github.com/rancher/cluster-api-provider-rke2/blob/0a2c0f84af2a7143ae8ee084d7d5ad8a8a0eda1c/bootstrap/api/v1beta1/rke2config_types.go#L178
+	// this would correspond to Metadata in linodego InstanceCreateOptions
+
+	// FirewallID is the id of the cloud firewall to apply to the Linode Instance
+	// +optional
+	FirewallID int `json:"firewallID,omitempty"`
+
+	// InterfaceGeneration is the generation of the interface to use for the cluster's
+	// nodes in interface / linodeInterface are not specified for a LinodeMachine.
+	// If not set, defaults to "legacy_config".
+	// +optional
+	// +kubebuilder:validation:Enum=legacy_config;linode
+	// +kubebuilder:default=legacy_config
+	InterfaceGeneration linodego.InterfaceGeneration `json:"interfaceGeneration,omitempty"`
+
+	// Interfaces is a list of legacy network interfaces to use for the instance.
+	// Conflicts with LinodeInterfaces.
+	// +optional
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
+	// +listType=atomic
+	Interfaces []linodego.InstanceConfigInterfaceCreateOptions `json:"interfaces,omitempty"`
+
+	// LinodeInterfaces is a list of Linode network interfaces to use for the instance. Requires Linode Interfaces beta opt-in to use.
+	// Conflicts with Interfaces.
+	// +optional
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
+	// +kubebuilder:object:generate=true
+	// +listType=atomic
+	LinodeInterfaces []linodego.LinodeInterfaceCreateOptions `json:"linodeInterfaces,omitempty"`
```

Comment:
unfortunately I think we might not be able to use linodego's structs for this since the DeepCopy functions don't exist / aren't generated